test/1.4/misc/T_dmllib.h

```diff
@@ -198,7 +198,8 @@ set_error_t stub_deserializer(attr_value_t *val, void *dest) {
           && SIM_attr_is_string(SIM_attr_list_item(*val, 0)))) {
         return Sim_Set_Illegal_Type;
     }
-    strncpy(dest, SIM_attr_string(SIM_attr_list_item(*val, 0)), 512);
+    strncpy(dest, SIM_attr_string(SIM_attr_list_item(*val, 0)), 511);
+    ((char *)dest)[511] = '\0';
```

Comment:
This is a reason not to use `strncpy`. I suggest `snprintf(dest, 512, "%s", ..)`, which does what you want. (note that gcc is smart and will optimize this approximately to strlen + memcpy)

Reply:
Noted. `strlcpy` is another alternative, but I didn't want to risk it not being available on some architecture.